appveyor.yml

```diff
@@ -3,52 +3,68 @@ build: off
 image: Visual Studio 2013
 cache:
   - '%LOCALAPPDATA%\pip\Cache'
-matrix:
-  fast_finish: true
 environment:
   global:
     WITH_ENV: 'cmd /E:ON /V:ON /C .\ci\run_with_env.cmd'
-  matrix:
-    - TOXENV: check
-      PYTHON_HOME: C:\Python36-x64
-      PYTHON_VERSION: 3.6
-      PYTHON_ARCH: 64
 
-    - TOXENV: docs
-      PYTHON_HOME: C:\Python36-x64
-      PYTHON_VERSION: 3.6
-      PYTHON_ARCH: 64
-
-    - TOXENV: '2.7,codecov'
-      PYTHON_HOME: C:\Python27
-      PYTHON_VERSION: 2.7
-      PYTHON_ARCH: 32
-
-    - TOXENV: '2.7,codecov'
-      PYTHON_HOME: C:\Python27-x64
-      PYTHON_VERSION: 2.7
-      PYTHON_ARCH: 64
-
-    - TOXENV: '3.6,codecov'
-      PYTHON_HOME: C:\Python36
-      PYTHON_VERSION: 3.6
-      PYTHON_ARCH: 32
-
-    - TOXENV: '3.6,codecov'
-      PYTHON_HOME: C:\Python36-x64
-      PYTHON_VERSION: 3.6
-      PYTHON_ARCH: 64
 init:
   - ps: echo $env:TOXENV
   - ps: echo "Python $env:PYTHON_VERSION (${env:PYTHON_ARCH}bit) from $env:PYTHON_HOME"
   - ps: ls C:\Python*
 install:
   - git submodule update --init --recursive
+test_script:
+  # -----
+  - set TOXENV=check
+  - set PYTHON_HOME=C:\Python36-x64
+  - set PYTHON_VERSION=3.6
+  - set PYTHON_ARCH=64
   - '%PYTHON_HOME%\Scripts\pip install --upgrade tox setuptools virtualenv wheel'
+  - '%WITH_ENV% %PYTHON_HOME%\Scripts\tox -v'
+
+  # check versions
   - '%PYTHON_HOME%\Scripts\virtualenv --version'
-  - '%PYTHON_HOME%\Scripts\pip --version'
   - '%PYTHON_HOME%\Scripts\tox --version'
-test_script:
+  - '%PYTHON_HOME%\Scripts\pip --version'
```

Comment:
i think generally we do the version checks _before_ running the test script (tox, in this case)... perhaps also grab these once per python_home...